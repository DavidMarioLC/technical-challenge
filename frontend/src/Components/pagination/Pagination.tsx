import './pagination.css'

interface Props {
  currentPage:number,
  totalPage:number,
  setCurrentPage:React.Dispatch<React.SetStateAction<number>>

}

const Pagination = ({currentPage,totalPage,setCurrentPage}: Props) => {

  const prevPage = () => {
    setCurrentPage(currentPage=>{
     if(currentPage === 1) return currentPage;

     return currentPage - 1
    })
  }

  const NextPage = () => {
    setCurrentPage(currentPage=>{
      if(currentPage === totalPage) return currentPage;
      return currentPage + 1
    })
  }

  
  const isDisabledPrevious = currentPage === 1 ? true : false;
  const isDisabledNext = currentPage === totalPage ? true : false;


  return (
    <div className="pagination">
        <button className={`btn btn--medium ${isDisabledPrevious?'btn--disabled':''}`} onClick={prevPage} >Previous</button>
         <p className="pagination__numbers">{currentPage}-{totalPage}</p>
        <button className={`btn btn--medium ${isDisabledNext?'btn--disabled':''}`}  onClick={NextPage} >Next</button>            
    </div>
  )
}

export default Pagination
