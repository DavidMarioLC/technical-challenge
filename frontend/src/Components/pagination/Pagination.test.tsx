import renderer from 'react-test-renderer';
import Pagination from './Pagination';


it('render Pagination correctly', () => {
  const tree = renderer.create(<Pagination
  currentPage={0}
  totalPage={0}
  setCurrentPage={()=>{}}
  
  />).toJSON();
  expect(tree).toMatchSnapshot();
});