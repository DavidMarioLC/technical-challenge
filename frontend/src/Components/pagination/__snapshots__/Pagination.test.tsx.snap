// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Pagination correctly 1`] = `
<div
  className="pagination"
>
  <button
    className="btn btn--medium "
    onClick={[Function]}
  >
    Previous
  </button>
  <p
    className="pagination__numbers"
  >
    0
    -
    0
  </p>
  <button
    className="btn btn--medium btn--disabled"
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
